google/oauth2/_client.py

```diff
@@ -129,32 +180,51 @@ def _token_endpoint_request_no_throw(
             if hasattr(response.data, "decode")
             else response.data
         )
-
-        if response.status == http_client.OK:
+        response_data = ""
+        try:
             # response_body should be a JSON
             response_data = json.loads(response_body)
-            break
-        else:
-            # For a failed response, response_body could be a string
-            try:
-                response_data = json.loads(response_body)
-                error_desc = response_data.get("error_description") or ""
-                error_code = response_data.get("error") or ""
-                if (
-                    any(e == "internal_failure" for e in (error_code, error_desc))
-                    and retry < 1
-                ):
-                    retry += 1
-                    continue
-            except ValueError:
-                response_data = response_body
-            return False, response_data
-
-    return True, response_data
+        except ValueError:
+            response_data = response_body
+        return response, response_data
+
+    response, response_data = _perform_request()
```

Comment:
There is some copy-paste that we can combine in a single method? 

Probably a "while True:" loop?

Reply:
I'll condense the code some more.

I am avoiding the while loop because I don't think the first request and the retries belong in the same loop. The iterator I have written will sleep on the first request. That would be bad :D